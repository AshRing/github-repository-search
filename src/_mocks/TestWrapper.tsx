import * as React from "react";

import { MemoryRouter } from "react-router";
import { ThemeProvider } from "styled-components";
import { theme } from "../_style";

const TestWrapper = ({ children }: { children: React.ReactChild }) => (
	<ThemeProvider theme={theme}>
		<MemoryRouter initialEntries={[{ pathname: "/", key: "head-test-key" }]}>
			{children}
		</MemoryRouter>
	</ThemeProvider>
);

export { TestWrapper };
