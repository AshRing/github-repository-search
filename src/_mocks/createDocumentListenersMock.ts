// used to test useClickOutside hook w/i components
export const createDocumentListenersMock = () => {
    const listeners = {};
    const handler = (domEl, event) => listeners?.[event]?.({ target: domEl });
    document.addEventListener = jest.fn((event, cb) => {
        listeners[event] = cb;
    });

    document.removeEventListener = jest.fn((event) => {
        delete listeners[event];
    });

    return {
        mouseDown: (domEl) => handler(domEl, "mousedown"),
        click: (domEl) => handler(domEl, "click"),
    };
};
