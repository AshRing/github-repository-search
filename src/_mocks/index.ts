export * from "./repo";
export * from "./TestWrapper";
export * from "./createDocumentListenersMock";
