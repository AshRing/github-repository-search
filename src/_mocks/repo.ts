import { IRepository } from "../_types";

export const mockRepo1: IRepository = {
	created_at: "10/01/2021",
	description: "",
	forks_count: 123,
	full_name: "@ash/test-project",
	homepage: "",
	html_url: "https://github.com",
	id: "1",
	language: "JavaScript",
	name: "test-project",
	open_issues_count: 0,
	owner: {
		login: "ash",
		id: "123",
		avatar_url: "some_url",
		url: "some_url",
	},
	stargazers_count: 55,
	updated_at: "10/03/2021",
	watchers_count: 12,
};
