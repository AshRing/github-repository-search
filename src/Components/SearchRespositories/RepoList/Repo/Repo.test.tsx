import * as React from "react";

import { Repo } from ".";
import { mockRepo1 } from "../../../../_mocks";
import { shallow } from "enzyme";

test("Repo matches snapshot", () => {
	const wrapper = shallow(<Repo repoInfo={mockRepo1} />);
	expect(wrapper).toMatchSnapshot();
});
