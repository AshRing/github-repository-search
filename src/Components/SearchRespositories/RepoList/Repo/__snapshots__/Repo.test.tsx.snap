// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo matches snapshot 1`] = `ShallowWrapper {}`;
